bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -247,11 +201,24 @@ class AsyncTlsConnection : public enable_shared_from_this<AsyncTlsConnection> {
         fs::path("server");
     _pSslContext->use_certificate_chain_file((path / fs::path("server.cert")
                                              ).string());
-    _pSslContext->use_private_key_file((path / fs::path("server.key")
+    _pSslContext->use_private_key_file((path / fs::path("pk.pem")
                                        ).string(),
                                        boost::asio::ssl::context::pem);
-    _pSslContext->use_tmp_dh_file((path / fs::path("dh2048.pem")
-                                  ).string());
+
+    // if we cant create EC DH params, it may mean that SSL version old or
+    // any other crypto related errors, we can't continue with TLS
```

Comment:
Well the key initialization is an integrated part of setting ssl server. This piece of code doesn't have any existence "rights" without being used specifically in setting up the server. I agree on making short functions, nevertheless, if the code is a part of the function main goal - it may reside there.